src/main/java/com/ericsson/ei/frontend/utils/BackEndInstancesUtils.java

```diff
@@ -16,134 +16,220 @@
+        JsonArray backEndList = new JsonArray();
+        for (BackEndInformation backendInformation : backEndInformationList) {
+            backEndList.add(backendInformation.getAsJsonObject());
+        }
+        return backEndList;
     }
 
-    public boolean checkIfInstanceAlreadyExist(JsonObject instance) {
-        for (JsonElement element : instances) {
-            if (element.getAsJsonObject().get(HOST).equals(instance.get(HOST)) &&
-                    element.getAsJsonObject().get(PORT).getAsInt() == instance.get(PORT).getAsInt() &&
-                    element.getAsJsonObject().get(PATH).equals(instance.get(PATH)) &&
-                    element.getAsJsonObject().get(NAME).equals(instance.get(NAME))) {
-                return true;
-            }
-        }
-        return false;
+    private void dumpBackEndInformationList() {
+        JsonArray jsonArrayToDump = parseBackEndsAsJsonArray();
+        backEndInstanceFileUtils.dumpJsonArray(jsonArrayToDump);
     }
 
-    public void writeIntoFile() {
+    private void parseBackEndInstances() {
         try {
-            Files.write(Paths.get(eiInstancesPath), instances.toString().getBytes());
-            parseBackEndInstancesFile();
+            JsonArray instances = backEndInstanceFileUtils.getInstancesFromFile();
+            backEndInformationList.clear();
+
+            for (JsonElement element : instances) {
+                backEndInformationList.add(new ObjectMapper().readValue(element.toString(), BackEndInformation.class));
+            }
+
+            ensureDefaultBackEnd();
         } catch (IOException e) {
-            LOG.error("Couldn't add instance to file " + e.getMessage());
+            LOG.error("Failure when trying to parse json " + e.getMessage());
         }
     }
 
-    private void parseBackEndInstancesFile() {
-        try {
-            information.clear();
-            instances = new JsonArray();
-            JsonArray inputBackEndInstances = new JsonParser().parse(new String(Files.readAllBytes(Paths.get(eiInstancesPath)))).getAsJsonArray();
-            for (JsonElement element : inputBackEndInstances) {
-                information.add(new ObjectMapper().readValue(element.toString(), BackEndInformation.class));
-                instances.add(element);
+    private void ensureDefaultBackEnd() {
+        if (defaultBackendInformation.getHost() == null ||
+                defaultBackendInformation.getPort() == null) {
+            LOG.debug("No default Host or Port set!");
+            return;
+        }
+        defaultBackendInformation.setDefaultBackend(true);
+        for (BackEndInformation information : backEndInformationList) {
+            if (defaultBackendInformation.getHost().equals(information.getHost()) &&
+                    defaultBackendInformation.getPort().equals(information.getPort()) &&
+                    information.isDefaultBackend()) {
+                LOG.debug("default backend already set!");
```

Comment:
I think "Default" in log will look better

Reply:
Default and back end :)